### PR TITLE
Add support for `Inhibit()`

### PR DESCRIPTION
**Pull request by d-loose**
- add `Inhibit()`, `Uninhibit()`, `isInhibited()` from `org.gnome.SessionManager` to `GnomeSessionManager` and `MateSessionManager`

### CODE REVIEW

**Comment by d-loose** [comment]
I might add this to the unified `UbuntuSession` class in a separate PR, if I can come up with a sensible way of doing so

File: lib/src/gnome_session_manager.dart [hunk 1]
```diff
@@ -18,6 +18,23 @@ enum GnomeLogoutMode {
   force,
 }
 
+enum GnomeInhibitionFlag {
+  /// Inhibit logging out
+  logout,
+
+  /// Inhbit user switching
```
Comment (jpnurmi) [inline lib/src/gnome_session_manager.dart:25]: typo: inh_bit

File: lib/src/gnome_session_manager.dart [hunk 2]
```diff
@@ -88,6 +105,37 @@ class GnomeSessionManager {
         .then((response) => response.values.first.asBoolean());
   }
 
+  /// Create an inhibition lock.
+  Future<int> inhibit(String appId, int topLevelXId, String reason,
+      Set<GnomeInhibitionFlag> flags) async {
```
Comment (jpnurmi) [inline lib/src/gnome_session_manager.dart:110]: nit-pick: the amount of positional arguments is getting high, could some of them be named?

File: lib/src/gnome_session_manager.dart [hunk 3]
```diff
@@ -88,6 +105,37 @@ class GnomeSessionManager {
...
+        .callMethod(
+            managerName,
+            'Inhibit',
+            [
+              DBusString(appId),
+              DBusUint32(topLevelXId),
+              DBusString(reason),
+              DBusUint32(encodeEnum(flags)),
+            ],
+            replySignature: DBusSignature('u'))
```
Comment (jpnurmi) [inline lib/src/gnome_session_manager.dart:121]: future "improvement" idea: use the pre-defined `DBusSignature.uint32` etc. constants to help with readability

File: lib/src/mate_session_manager.dart [hunk 4]
```diff
@@ -18,6 +18,23 @@ enum MateLogoutMode {
   force,
 }
 
+enum MateInhibitionFlag {
+  /// Inhibit logging out
+  logout,
+
+  /// Inhbit user switching
```
Comment (jpnurmi) [inline lib/src/mate_session_manager.dart:25]: inh_bit

**Review by jpnurmi — APPROVED** [review]
1337! :+1:

File: lib/src/mate_session_manager.dart (hunk 4 above)
Comment (d-loose) [reply lib/src/mate_session_manager.dart:25]: :+1:

File: lib/src/gnome_session_manager.dart (hunk 2 above)
Comment (d-loose) [reply lib/src/gnome_session_manager.dart:110]: :+1:

File: lib/src/gnome_session_manager.dart (hunk 1 above)
Comment (d-loose) [reply lib/src/gnome_session_manager.dart:25]: :+1:

File: lib/src/gnome_session_manager.dart (hunk 3 above)
Comment (d-loose) [reply lib/src/gnome_session_manager.dart:121]: Thanks, I'll keep it in mind - I need to revisit all of those DBus libraries and improve all those things I've learned :)

**Pull request merged**